src/transformers/trainer.py

```diff
@@ -1140,7 +1141,28 @@ def call_model_init(self, trial=None):
 
         return model
 
-    def _wrap_model(self, model, training=True):
+    def torch_jit_model(self, model, training=False, dataloader=None):
+        if not training:
+            jit_inputs = []
+            example_batch = list(dataloader)[0]
```

Comment:
No, we can't unpack the whole dataloader as it might be very slow and make the user run out of RAM if the dataset is huge.
```suggestion
            for example_batch in dataloader:
                break
```